smdebug/tensorflow/keras.py

```diff
@@ -390,6 +394,30 @@ def _save_custom_tensors_post_step(self):
             self._save_tensor_to_file(tensor_name, tensor_value, collection_names)
         self.custom_tensors_to_save.clear()
 
+    def should_save_layer(self, layer_name):
+        # Called in AWS TF to determine
+        # if a particular layer value
+        # should be saved
+        if self._is_collection_being_saved_for_step(CollectionKeys.LAYERS):
+            return True
+        collections_to_save = self._get_collections_to_save_for_step()
+        for c in collections_to_save:
+            if match_inc(layer_name, c.include_regex):
+                return True
+        return False
+
+    def should_save_gradient(self, layer_name):
```

Comment:
can you combine this with should_save_layer() ? the definition is the same except for the collection name

Reply:
We can reduce some duplication here for sure.
But the idea behind two functions was to eliminate the need to use strings or string-constants in AWS TF, since we do not have the CollectionKeys Enum defined in their codebase.